src/App/App.js

```diff
@@ -1,40 +1,49 @@
-import React, { Component } from "react";
+import React, { useEffect, useState } from "react";
 import styles from "./App.module.scss";
 import { connect } from "react-redux";
 import RetailerView from "../views/RetailerView";
 import UserView from "../views/UserView";
-import RetailerProfileView from '../views/RetailerProfileView';
-import UserProfileView from '../views/UserProfileView';
-import { Switch, Route } from 'react-router-dom';
+import RetailerProfileView from "../views/RetailerProfileView";
+import UserProfileView from "../views/UserProfileView";
+import { Switch, Route } from "react-router-dom";
 
-class App extends Component {
-  render() {
-    return (
-      <div className={styles.App}>
-        <div className={styles.appMobileContainer}>
-          <Switch>
-            <Route
-              path='/userview'
-              component={UserView} />
-            <Route
-              path='/retailerview'
-              component={RetailerView} />
-            <Route
-              path='/retailerprofile'
-              component={RetailerProfileView} />
-            <Route
-              path='/userprofile'
-              component={UserProfileView} />
-          </Switch>
-        </div>
+import AuthView from "../views/AuthView";
+
+var App = (props) => {
+  const [retailerLoggedIn, setRetailerLoggedIn] = useState(false);
+
+  useEffect(() => {
+    let lsRetailer = localStorage.getItem("retailer");
+
+    if (!lsRetailer) {
+      setRetailerLoggedIn(false);
+    } else {
+      setRetailerLoggedIn(true);
+    }
+  }, [props.isLoggedIn]);
+
+  return (
+    <div className={styles.App}>
+      <div className={styles.appMobileContainer}>
+        <Switch>
+          <Route path="/userview" component={UserView} />
+          <Route
```

Comment:
I found that you can pass props by just adding it to the Route component, so Idk if you want to revert the removal of the useEffect in the AuthView because currently the state doesn't change when someone logs in, causing the app to stay on the login page after successful login.

Reply:
what does that look like, can you send me an example?

Reply:
so right after line 32 you would put
```
setRetailerLoggedIn={setRetailerLoggedIn}
```